packages/plugin-ext/src/hosted/node/hosted-plugin.ts

```diff
@@ -51,11 +53,13 @@ export class HostedPluginSupport {
     }
 
     setClient(client: HostedPluginClient): void {
+        this.client = client;
         this.hostedPluginProcess.setClient(client);
         this.pluginRunners.forEach(runner => runner.setClient(client));
     }
 
     clientClosed(): void {
+        this.isPluginProcessRunning = false;
```

Comment:
I don't see when clientClosed is called ?

Reply:
@benoitf [there](https://github.com/theia-ide/theia/blob/8ca8c80044001c20c20a36384de4e3202f541388/packages/plugin-ext/src/hosted/node/plugin-service.ts#L50)

Reply:
yes but https://github.com/theia-ide/theia/blob/master/packages/plugin-ext/src/hosted/node/plugin-ext-hosted-backend-module.ts#L50

Reply:
what is happening if you refresh the browser ?

Reply:
When set new client we restart plugin node: https://github.com/theia-ide/theia/blob/8ca8c80044001c20c20a36384de4e3202f541388/packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts#L42

Reply:
ah yes